src/app/core/connection.ts

```diff
@@ -541,45 +539,45 @@ export default class Connection {
      */
     regenerateSegments() {
         const pathscopy = this.getPaths();
-        const ret = [];
+        const ret: Segment = [];
         let waypointscopy;
         for (const j in pathscopy) {
             waypointscopy = pathscopy[j];
             for (let i = 0; i < waypointscopy.length - 1; i++) {
-                const segment = [waypointscopy[i], waypointscopy[i + 1]];
+                const segment: Point = [waypointscopy[i], waypointscopy[i + 1]];
                 ret.push(segment);
             }
         }
         this.updateSegments(ret);
     }
 
-    /**
-     * Allows the user to set the source of the connection
-     * @param {Object} component
-     * @param {ComponentPort} port
-     * @memberof Connection
-     * @returns {void}
-     */
-    setSource(component, port) {
-        if (typeof component !== "string" && !(component instanceof String)) {
-            console.error("The reference object value can only be a string");
-        }
-        this.__source = new ConnectionTarget(component, port);
-    }
-
-    /**
-     * Allows the user to add a sink to the connection
-     * @param {string} component
-     * @param {ComponentPort} port
-     * @memberof Connection
-     * @returns {void}
-     */
-    addSink(component, port) {
-        if (typeof component !== "string" || !(component instanceof String)) {
-            console.error("The reference object value can only be a string");
-        }
-        this.__sinks.push(new ConnectionTarget(component, port));
-    }
+    // /**
+    //  * Allows the user to set the source of the connection
+    //  * @param {Object} component
+    //  * @param {ComponentPort} port
+    //  * @memberof Connection
+    //  * @returns {void}
+    //  */
+    // setSource(component: string, port: ComponentPort) {
+    //     if (typeof component != "string" && !(component instanceof String)) {
+    //         console.error("The reference object value can only be a string");
+    //     }
+    //     this._source = new ConnectionTarget(component, port);
+    // }
+
+    // /**
+    //  * Allows the user to add a sink to the connection
+    //  * @param {string} component
+    //  * @param {ComponentPort} port
+    //  * @memberof Connection
+    //  * @returns {void}
+    //  */
+    // addSink(component: string, port: ComponentPort) {
+    //     if (typeof component != "string" || !(component instanceof String)) {
+    //         console.error("The reference object value can only be a string");
```

Comment:
Are there setters for `sinks` and `sources`?

Reply:
Not with those functions commented out. addSink is never used and setSource only appears in a min.js file

Reply:
We can leave this be for now, I have a feeling we will need to revisit this